### ISSUE TITLE
False positive "Redundant use of assertTrue with constant value False"

### ISSUE THREAD

**Issue by jwilk** [PyCQA/pylint#956]
### Current behavior

pylint emits "Redundant use of assertTrue" even when the constant is false, i.e. the assert is definitely not redundant.

Minimal example:
```python
from nose.tools import assert_true
assert_true(False)
```

Real-world example:
https://github.com/jwilk/didjvu/blob/0.8/tests/test_version.py#L46

### Expected behavior

pylint emits "Redundant use of assertTrue" only when the constant is true.

### pylint --version output
```
No config file found, using default configuration
pylint 1.5.6, 
astroid 1.4.6
Python 2.7.12rc1 (default, Jun 13 2016, 09:20:59) 
[GCC 5.4.0 20160609]
```

**Comment by PCManticore** [comment]
I'm closing this, since it is the expected behavior.

The problem with this code is that the condition will always be the same, leading to the same test result all the time, either passing or failing. This in turn might suggest a broken test, for instance if the user wanted to use assertEqual / assert_equal instead, but wrote assertTrue / assert_true instead, which is why the checker emits for both the cases of the first argument being False or True. This does not happen, for example, if the argument was a variable, which might suggest to Pylint that it was an intentional use.

Probably, in your example, there has to be a better way exported by nose for failing a test (a fail method perhaps?). Alternatively, you can just disable the check on that line.

**Comment by lleeoo** [comment]
Here is a usecase:

```python
for i in range(5):
    attempt()
else:
    self.assertTrue(False), "The heuristic failed to find a solution in 5 tries."
```

Sure, you could add a different condition. Can you suggest one that would be less obtuse?

**Comment by sam-s** [comment]
What is the right way to mark unreachable code in tests?
The only thing I could come up with so far is
```
try:
    ... # code that must raise
    self.assertTrue(False)  # pylint: disable=redundant-unittest-assert
except ...:
    ....
```

**Comment by davmlaw** [comment]
```
self.fail("The heuristic failed to find a solution in 5 tries.")
```
